pkg/cri/server/container_create_linux.go

```diff
@@ -260,6 +261,15 @@ func (c *criService) containerSpec(
 
 	supplementalGroups := securityContext.GetSupplementalGroups()
 
+	// Get RDT class
```

Comment:
do we need a config flag to ignoreRdtNotEnabledErrors default false..

Reply:
if you go the cri inited and configured route ^ the ignore flag and config path string can be added to the cri config as well

Reply:
I added a patch introducing `cri.containerd.ignore_rdt_not_enabled_errors`config flag. I felt cri plugin was the appropriate place for this config flag. Maybe this patch should be split into a separate PR(?)

Reply:
one is fine.. because it's related imo.. (note maintainers don't have squash merge on this repo.., someone might ask for the commits to be squashed around/after review approvals)

Reply:
> one is fine..because it's related imo..

👍  I was pondering a separate PR for the case this new patch creates a lot of debate. I.e. basically removing extra friction by splitting it out. But let's keep it here, for now

> (note maintainers don't have squash merge on this repo.., someone might ask for the commits to be squashed around/after review approvals)

Sure, I can do that if asked. My personal style is just splitting changes into discrete, logically and functionally self-contained commits. In the hope of helping understand the in-code dependencies, making it easier to read for reviewers and also future readers (who are looking at the commit history)

Reply:
Should we just ignore RDT if its not supported/enabled? `IgnoreRdtNotEnabledErrors` feels hacky.

Reply:
> RDT is for cache and memory bandwidth management. Is it worth failing?

I guess it really depends on the usage scenario. There are use cases with high SLA/QoS requirements where you certainly want to fail if it fails. I agree that `IgnoreRdtNotEnabledErrors` sounds hackish, maybe changing the name could ease the pain?

Reply:
`IgnoreRdtErrors` may be?

Reply:
`IgnoreRdtErrors` would be semantically slightly different, i.e. I think with this we should ignore all rdt-related errors (that happen in the cri plugin). Whereas `IgnoreRdtNotEnabledErrors` now ignores errors only in the case that rdt has not been enabled (in containerd scope) – i.e. errors are not ignored if rdt has been (successfully) enabled by the sysadmin. I'm basically fine with either one.

In practice, this is related to the annotation parsing that happens now on the runtime side. Going forward, my goal is to get rdt in the cri protocol, after which rdt prolly changes into a "pass-through mode" where the client sets the rdt class in the containercreate request and the cri plugin just passes this into the oci runtime spec. Errors will happen in runc side.